File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/CodyTaskState.kt
package com.sourcegraph.cody.agent.protocol

import com.google.gson.annotations.SerializedName

enum class CodyTaskState {
  @SerializedName("Idle") Idle,
  @SerializedName("Working") Working,
  @SerializedName("Inserting") Inserting,
  @SerializedName("Applying") Applying,
  @SerializedName("Formatting") Formatting,
  @SerializedName("Applied") Applied,
  @SerializedName("Finished") Finished,
  @SerializedName("Error") Error,
  @SerializedName("Pending") Pending
}

File: agent/bindings/csharp/CodyTaskState.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{

  public enum CodyTaskState
  {
    [EnumMember(Value = "Idle")]
    Idle,
    [EnumMember(Value = "Working")]
    Working,
    [EnumMember(Value = "Inserting")]
    Inserting,
    [EnumMember(Value = "Applying")]
    Applying,
    [EnumMember(Value = "Applied")]
    Applied,
    [EnumMember(Value = "Finished")]
    Finished,
    [EnumMember(Value = "Error")]
    Error,
    [EnumMember(Value = "Pending")]
    Pending,
  }
}
